tests/integration/test_consume_flow.py

```diff
@@ -4,53 +4,35 @@
 #
 import os
 import shutil
+from typing import Union
 
 import pytest
+from web3 import Web3
 
 from ocean_lib.agreements.service_types import ServiceTypes
+from ocean_lib.config import Config
 from ocean_lib.data_provider.data_service_provider import DataServiceProvider
 from ocean_lib.models.erc20_token import ERC20Token
-from ocean_lib.models.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.erc721_nft import ERC721NFT
 from ocean_lib.ocean.ocean_assets import OceanAssets
+from ocean_lib.structures.file_objects import IpfsFile, UrlFile
 from ocean_lib.web3_internal.constants import ZERO_ADDRESS
 from ocean_lib.web3_internal.currency import to_wei
+from ocean_lib.web3_internal.wallet import Wallet
 from tests.resources.ddo_helpers import get_first_service_by_type
-from tests.resources.helper_functions import get_address_of_type
 
 
 @pytest.mark.integration
-def test_consume_flow(web3, config, publisher_wallet, consumer_wallet, file1):
-    erc721_factory_address = get_address_of_type(
-        config, ERC721FactoryContract.CONTRACT_NAME
-    )
-    erc721_factory = ERC721FactoryContract(web3, erc721_factory_address)
-
-    # Publisher deploys NFT contract
-    tx = erc721_factory.deploy_erc721_contract(
-        name="NFT1",
-        symbol="NFTSYMBOL",
-        template_index=1,
-        additional_metadata_updater=ZERO_ADDRESS,
-        additional_erc20_deployer=ZERO_ADDRESS,
-        token_uri="https://oceanprotocol.com/nft/",
-        transferable=True,
-        owner=publisher_wallet.address,
-        from_wallet=publisher_wallet,
-    )
-    tx_receipt = web3.eth.wait_for_transaction_receipt(tx)
-    registered_event = erc721_factory.get_event_log(
-        ERC721FactoryContract.EVENT_NFT_CREATED,
-        tx_receipt.blockNumber,
-        web3.eth.block_number,
-        None,
-    )
-
-    assert registered_event[0].event == "NFTCreated"
-    assert registered_event[0].args.admin == publisher_wallet.address
-    erc721_address = registered_event[0].args.newTokenAddress
-
+def test_consume_flow(
+    web3: Web3,
+    config: Config,
+    publisher_wallet: Wallet,
+    consumer_wallet: Wallet,
+    erc721_nft: ERC721NFT,
+    file1: Union[UrlFile, IpfsFile],
```

Comment:
Isn't there a superclass to these two file types? If not, there should be. Otherwise when we add a new file type we'll have to update the typings everywhere.

Reply:
There is not a superclass.  I can add one.

Reply:
Done.